src/hotspot/share/gc/shenandoah/shenandoahSharedVariables.hpp

```diff
@@ -236,6 +236,13 @@ struct ShenandoahSharedEnumFlag {
     return (T)Atomic::cmpxchg(&value, (ShenandoahSharedValue)expected, (ShenandoahSharedValue)new_value);
   }
 
+  T xchg(T new_value) {
+    assert (new_value >= 0, "sanity");
+    assert (new_value < (sizeof(ShenandoahSharedValue) * CHAR_MAX), "sanity");
+    // Hmm, no platform template specialization defined for exchanging one byte... (up cast to intptr is workaround).
+    return (T)Atomic::xchg((intptr_t*)&value, (intptr_t)new_value);
```

Comment:
That... likely gets awkward on different endianness. See the complicated dance `Atomic::CmpxchgByteUsingInt` has to do to handle it. 

Not to mention we are likely writing to adjacent memory location. Which is _currently_ innocuous, since we hit padding, but it is not very reliable.

Reply:
`PlatformCmpxchg` has specializations on aarch64 and x86 for `sizeof(T) == 1`. Should we also add platform specializations for `PlatformXchg` for `sizeof(T) == 1`? (It has them for `4` and `8`). Could also do what `XchgUsingCmpxchg` does...

Reply:
Maybe it is easiest/safest to change declaration of value to intptr_t.